lib/fauna_model/base.rb

```diff
@@ -0,0 +1,269 @@
+module Fauna
+  module Model
+    class Base
+      include ActiveModel::Validations
+      include Fauna::Model::Dirty
+
+      class << self
+        def fauna_class
+          @fauna_class || fail(InvalidClass.new('No class set'))
+        end
+
+        def fauna_class=(fauna_class)
+          fauna_class = Fauna::Ref.new(fauna_class.to_s) unless fauna_class.is_a? Fauna::Ref
+          @fauna_class = fauna_class
+
+          field 'ref', 'ref', internal_readonly: true
+          field 'ts', 'ts', internal_readonly: true
+          field 'fauna_class', 'class', internal_readonly: true
+
+          case fauna_class
```

Comment:
I'll look into the ability to subclass models, I don't think I've tested doing that yet.